### DISCUSSION TITLE
Why are the viewport x/y coordinates negative after I zoom (codesandbox provided)

### DISCUSSION THREAD

**Discussion by raarts** [xyflow/xyflow#2104] [Q&A]
Please see this codesandbox: https://codesandbox.io/s/determined-rumple-vp9nvo?file=/src/App.tsx
If you click in the flow, to focus, and then press 'i', it zooms in.

I think of the canvas as a piece of paper, with 0,0 at the left top.
Using setViewport(), the viewport (meaning the ReactFlow component) can be moved across that paper using the coordinates, and in addition the paper itself can be moved closer of farther away (zoom in and out).  

I expected the viewport coordinates after zooming to be  > 0, but they are < 0.

Panning shows the same, so my understanding must be completely off. Can anybody help me out?

**Comment by bcakmakoglu** [comment]
`x` and `y` correspond to a transform value that is applied to the viewport to achieve a zoom & pan effect. (check the devtools for the `.react-flow__viewport` element)
So when you're moving the pane to the bottom right, instead of getting `x > 0` or `y > 0` you're getting the opposite as the viewport is transformed by  `-x` or `-y` (again, when moving to the bottom right - it's vice versa for top left).
`x: -275, y: -333, zoom: 1.44` would become `transform: translate(-275px, -333px) scale(1.44);`

Generally you can assume that movement to the bottom right will be negative deltas and movement to top left positive deltas.

Maybe that helps

**Comment by raarts** [comment]
Thanks for responding, I find this to be counterintuitive, and have lots of trouble wrapping my mind around it.

I find it easy to think of the canvas as a piece of paper with coordinates (0,0) at the left top. If I would want to set a viewport, I would imagine I would specify the left/top of the viewport in canvas coordinates, so (100, 100) would be a bit to the right and down.
I also would want to specify the width and height of the viewport in coordinates or alternatively as a percentage of the canvas width. Both would automatically imply a zoom factor.

In your last sentence:

> Generally you can assume that movement to the bottom right will be negative deltas and movement to top left positive deltas.

do you mean moving the pane around, or moving the viewport around?

(Now I think about it, part of the confusion might be that as a user, I need to drag _the pane_ to move the viewport over the pane, but as a programmer I need to use setViewport(), so moving the _viewport_  to move the viewport over the pane.)

About the translate part, I understand this is used to translate screen coordinates into canvas coordinates. I think the docs would benefit greatly if these concepts were better explained. 

Sorry to put out a blurb of text, but all this has been a major hurdle for my development.
